### PR TITLE
Add system tests for chain upgrade - backport

### PR DESCRIPTION
**Pull request by alpe**
Backport #1643

### CODE REVIEW

**Review by pinosu — APPROVED** [review]
Great work! 🥇 LGTM 👍 
There are still some todos in the test upgrade code.. not sure if they can be removed or you were planing to add more maybe in another pr.

File: Makefile [hunk 1]
```diff
@@ -161,22 +161,22 @@ test-system: install
...
+lint:
 	golangci-lint run --tests=false
 	find . -name '*.go' -type f -not -path "./vendor*" -not -path "./tests/system/vendor*" -not -path "*.git*" -not -path "*_test.go" | xargs gofumpt -d
 
-format: format-tools
+format:
 	find . -name '*.go' -type f -not -path "./vendor*" -not -path "./tests/system/vendor*" -not -path "*.git*" -not -path "./client/lcd/statik/statik.go" | xargs gofumpt -w
 	find . -name '*.go' -type f -not -path "./vendor*" -not -path "./tests/system/vendor*" -not -path "*.git*" -not -path "./client/lcd/statik/statik.go" | xargs misspell -w
-	find . -name '*.go' -type f -not -path "./vendor*" -not -path "./tests/system/vendor*" -not -path "*.git*" -not -path "./client/lcd/statik/statik.go" | xargs goimports -w -local github.com/CosmWasm/wasmd
+	find . -name '*.go' -type f -not -path "./vendor*" -not -path "./tests/system/vendor*" -not -path "*.git*" -not -path "./client/lcd/statik/statik.go" | xargs gci write --skip-generated -s standard -s default -s "prefix(cosmossdk.io)" -s "prefix(github.com/cosmos/cosmos-sdk)" -s "prefix(github.com/CosmWasm/wasmd)" --custom-order
```
Comment (pinosu) [inline Makefile:173]: 👍

File: tests/system/upgrade_test.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+	t.Log("waiting for upgrade info")
+	sut.AwaitUpgradeInfo(t)
+	sut.StopChain()
+
+	t.Log("Upgrade height was reached. Upgrading chain")
+	sut.ExecBinary = currentBranchBinary
+	sut.StartChain(t)
+
+	t.Skip("wasmvm 1.4 upgrade fails, currently. Skipping for now")
```
Comment (pinosu) [inline tests/system/upgrade_test.go:93]: 👍

**Review by pinosu — APPROVED** [review]
Very nice! 💯  LGTM 👍

File: tests/system/cli.go [hunk 3]
```diff
@@ -381,6 +389,46 @@ func (c WasmdCli) IsInTendermintValset(valPubKey cryptotypes.PubKey) (rpc.Result
 	return valResult, found
 }
 
+// SubmitUpgradeGovProposal submit a chain upgrade gov v1 proposal
```
Comment (pinosu) [inline tests/system/cli.go:392]:
```suggestion

```

**Review by pinosu — APPROVED** [review]
LGTM! 👍

**Comment by alpe** [comment]
The upgrade test is flacky on this branch. Not sure if we should invest more time into this. 🤷

**Pull request closed without merging**